commands/pack.go

```diff
@@ -56,24 +56,37 @@ func init() {
 }
 
 func packRun(flags packFlags) error {
-	buildpackDir, err := os.MkdirTemp("", "dup-dest")
+
+	if flags.buildpackTOMLPath == "" && flags.extensionTOMLPath == "" {
+		return fmt.Errorf(`"buildpack" or "extension" flag is required`)
+	}
+
+	tmpDir, err := os.MkdirTemp("", "dup-dest")
 	if err != nil {
 		return fmt.Errorf("unable to create temporary directory: %s", err)
 	}
-	defer os.RemoveAll(buildpackDir)
+	defer os.RemoveAll(tmpDir)
+
+	if flags.extensionTOMLPath != "" {
+		err := packRunExtension(flags, tmpDir)
+		if err != nil {
+			return fmt.Errorf("failed to pack extension: %s", err)
+		}
+		return nil
+	}
 
 	directoryDuplicator := cargo.NewDirectoryDuplicator()
-	err = directoryDuplicator.Duplicate(filepath.Dir(flags.buildpackTOMLPath), buildpackDir)
+	err = directoryDuplicator.Duplicate(filepath.Dir(flags.buildpackTOMLPath), tmpDir)
 	if err != nil {
 		return fmt.Errorf("failed to duplicate directory: %s", err)
 	}
 
-	flags.buildpackTOMLPath = filepath.Join(buildpackDir, filepath.Base(flags.buildpackTOMLPath))
+	buildpackTOMLPath := filepath.Join(tmpDir, filepath.Base(flags.buildpackTOMLPath))
 
 	configParser := cargo.NewBuildpackParser()
-	config, err := configParser.Parse(flags.buildpackTOMLPath)
+	config, err := configParser.Parse(buildpackTOMLPath)
 	if err != nil {
-		return fmt.Errorf("failed to parse buildpack.toml: %s", err)
+		return fmt.Errorf("failed to parse .toml file : %s", err)
```

Comment:
It might be good to a print the file name here instead of just `.toml file`.

E.g.:

```
failed to parse buildpack.toml
```

or:

```
failed to parse extension.toml
```